src/gui/guiFormSpecMenu.cpp

```diff
@@ -4570,7 +4593,8 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 		case BET_OTHER: {
 			// Some other mouse event has occured
 			// Currently only left-double-click should trigger this
-			if (!s.isValid() || event.MouseInput.Event != EMIE_LMOUSE_DOUBLE_CLICK)
+			if (!s.isValid() || (event.EventType == EET_MOUSE_INPUT_EVENT &&
```

Comment:
Because a "second touch" event should never be able to trigger a "double-click pick-up all", the condition should look like this:

```suggestion
			if (!s.isValid() || event.EventType != EET_MOUSE_INPUT_EVENT ||
					event.MouseInput.Event != EMIE_LMOUSE_DOUBLE_CLICK)
```